apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex

```diff
@@ -33,43 +59,309 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
-              <button type="button" class="btn btn--large" phx-click="cancel-edit">Cancel</button>
-            </div>
+            <div class="edit-button-group form-grid">
+              <div class="edit-button-group__cancel-btn">
+                <a href="#metadata-form" id="back-button" class="btn btn--back btn--large" phx-click="toggle-component-visibility" phx-value-component-expand="metadata_form">Back</a>
+                <button type="button" class="btn btn--large" phx-click="cancel-edit">Cancel</button>
+              </div>
 
-            <div class="edit-button-group__save-btn">
-              <a href="#url-form" id="next-button" class="btn btn--next btn--large btn--action" phx-click="toggle-component-visibility" phx-value-component-expand="url_form" phx-value-component-collapse="data_dictionary_form">Next</a>
-              <%= submit("Save Draft", id: "save-button", name: "save-button", class: "btn btn--save btn--large", phx_value_action: "draft") %>
+              <div class="edit-button-group__save-btn">
+                <a href="#url-form" id="next-button" class="btn btn--next btn--large btn--action" phx-click="toggle-component-visibility" phx-value-component-expand="url_form">Next</a>
+                <button id="save-button" name="save-button" class="btn btn--save btn--large" type="button" phx-click="camsave">Save Draft</button>
+              </div>
             </div>
           </div>
-        </div>
+        </form>
       </div>
+
+      <%= live_component(@socket, AndiWeb.EditLiveView.DataDictionaryAddFieldEditor, id: :data_dictionary_add_field_editor, eligible_parents: get_eligible_data_dictionary_parents(@dataset), visible: @add_data_dictionary_field_visible, dataset_id: @dataset.id,  selected_field_id: @selected_field_id ) %>
+
+      <%= live_component(@socket, AndiWeb.EditLiveView.DataDictionaryRemoveFieldEditor, id: :data_dictionary_remove_field_editor, selected_field: @current_data_dictionary_item, visible: @remove_data_dictionary_field_visible) %>
     </div>
     """
   end
+
+  def handle_event("cam", %{"data_dictionary_form_schema" => form_schema}, socket) do
+    form_schema
+    |> DataDictionaryFormSchema.changeset_from_form_data()
+    |> complete_validation(socket)
+  end
+
+  def handle_event("camsave", _, socket) do
+    changeset =
+      socket.assigns.changeset
+      |> Map.put(:action, :update)
+
+    AndiWeb.Endpoint.broadcast_from(self(), "form-save", "form-save", %{form_changeset: changeset})
+
+    new_validation_status =
+      case changeset.valid? do
+        true -> "valid"
+        false -> "invalid"
+      end
+
+    {:noreply, assign(socket, changeset: changeset, validation_status: new_validation_status)}
+  end
+
+  def handle_event("toggle-component-visibility", %{"component-expand" => next_component}, socket) do
```

Comment:
Is there any way to dry up these repetitive visibility toggles? Like with a macro or import?